grakn-dist/src/main/java/ai/grakn/dist/DistGrakn.java

```diff
@@ -0,0 +1,719 @@
+
+        String pid = processHandler.getPidFromPsOf(Grakn.class.getName());
+
+        try {
+            Files.write(Paths.get(GRAKN_PID),pid.getBytes(StandardCharsets.UTF_8));
+        } catch (IOException e) {
+            System.out.println("Cannot write Grakn PID on a file");
+        }
+
+        LocalDateTime init = LocalDateTime.now();
+        LocalDateTime timeout = init.plusSeconds(GRAKN_STARTUP_TIMEOUT_S);
+
+        while(LocalDateTime.now().isBefore(timeout)) {
+            System.out.print(".");
+            System.out.flush();
+
+            String host = graknConfig.getProperty(GraknConfigKey.SERVER_HOST_NAME);
+            int port = graknConfig.getProperty(GraknConfigKey.SERVER_PORT);
+
+            if(graknIsRunning() && graknCheckIfReady(host,port,REST.WebPath.STATUS)) {
+                System.out.println("SUCCESS");
+                graknIsStarted =true;
+                return;
+            }
+            try {
+                Thread.sleep(WAIT_INTERVAL_S * 1000);
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+        }
+
+        if(Files.exists(Paths.get(GRAKN_PID))) {
+            try {
+                Files.delete(Paths.get(GRAKN_PID));
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+
+        System.out.println("FAILED!");
+        System.out.println("Unable to start Grakn");
+    }
+
+    private boolean graknCheckIfReady(String host, int port, String path) {
+        try {
+            URL siteURL = UriBuilder.fromUri(new SimpleURI(host, port).toURI()).path(path).build().toURL();
+            HttpURLConnection connection = (HttpURLConnection) siteURL
+                    .openConnection();
+            connection.setRequestMethod("GET");
+            connection.connect();
+
+            int code = connection.getResponseCode();
+            if (code == 200) {
+                return true;
+            } else {
+                return false;
+            }
+        } catch (IOException e) {
+            return false;
+        }
+    }
+
+    private void startAll() {
+        startStorage();
+        startQueue();
+        startGrakn();
+
+        if(!graknIsStarted || !queueIsStarted || !storageIsStarted) {
+            System.out.println("Please run 'grakn server status' or check the logs located under 'logs' directory.");
+        }
+
+    }
+
+    public void graknStopProcess() {
+        String graknPid="";
+        if(Files.exists(Paths.get(GRAKN_PID))) {
+            try {
+                graknPid = new String(Files.readAllBytes(Paths.get(GRAKN_PID)),StandardCharsets.UTF_8);
+                graknPid = graknPid.trim();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+            if (graknPid.trim().isEmpty()) {
+                return;
+            }
+            processHandler.kill(graknPid);
+        }
+
+        while(true) {
```

Comment:
We could use the seldom-seen `do .. while` loop in these places (and the other `stop` methods):

```
do {
    ...
} while (outputCommand.exitStatus != 0);

System.out.println("SUCCESS");
```